client/src/key_management/bundle.rs

```diff
@@ -53,4 +55,230 @@ impl KeyBundleContent {
             None => Ok(pre_key_bundle),
         }
     }
+
+    pub fn serialize(&self) -> PrimitiveKeyBundleContent {
+        let one_time_key: (u32, Box<[u8]>) = if (self.onetime_public_key.is_none()) {
+            (0, Box::new([]))
+        } else {
+            let (prekey_id, public_key) = self.onetime_public_key.expect("");
+            (prekey_id.into(), public_key.serialize())
+        };
+
+        let kyber_key: (u32, Box<[u8]>, Vec<u8>) = if (self.kyper_key_essentials.is_none()) {
+            (0, Box::new([]), vec![])
+        } else {
+            let (kyber_id, kyber_key, vec) = self.kyper_key_essentials.to_owned().expect("");
+            (kyber_id.into(), kyber_key.serialize(), vec)
+        };
+
+        PrimitiveKeyBundleContent::new(
+            self.registration_id,
+            self.device_id.into(),
+            Some(one_time_key.0),
+            Some(one_time_key.1),
+            self.signed_public_key_id.into(),
+            self.signed_public_key.serialize(),
+            self.signed_signature.clone(),
+            self.identity_key.serialize(),
+            Some(kyber_key.0),
+            Some(kyber_key.1),
+            Some(kyber_key.2),
+        )
+    }
+
+    pub fn deserialize(bundle: PrimitiveKeyBundleContent) -> KeyBundleContent {
+        let one_time_key: Option<(PreKeyId, PublicKey)> =
+            if (bundle.onetime_public_key_id.is_none()
+                || bundle.onetime_public_key.is_none()
+                || bundle
+                    .onetime_public_key_id
+                    .expect("One time pre key id is somehow None after a none check - kill me")
+                    == 0)
+            {
+                None
+            } else {
+                Some((
+                    bundle.onetime_public_key_id.expect("").into(),
```

Comment:
Use unwrap instead of expect if no custom error message is chosen anyway

Reply:
Mikkel er en unwrap hater så nu er der expect i stedet for